lib/rack/session/encryptor.rb

```diff
@@ -327,6 +327,21 @@ def message_secret_from_salt(salt)
         def set_cipher_key(cipher, key)
           cipher.key = key
         end
+
+        if RUBY_ENGINE == 'jruby'
+          # JRuby's OpenSSL implementation doesn't currently support passing
+          # an argument to #auth_tag. Here we work around that.
+          def auth_tag_from(cipher)
+            tag = cipher.auth_tag
+            raise Error, 'the auth tag must be 16 bytes long' if tag.bytesize != 16
```

Comment:
Would it make more sense to use `warn` here rather than blowing up completely?

Reply:
@headius do we need to do this or can we just assume `auth_tag` will always return 16 bytes (by default).

Reply:
Ahh I think @enebo might have pinged me about this. I am not sure... I'm guessing if the size is off we're not properly terminating a string or something.

Reply:
I don’t think there is anything wrong with the size… just wondering if we can depend on the default.

Reply:
> Would it make more sense to use `warn` here rather than blowing up completely?

While encoding will work, as it doesn't depend on the size of the auth tag, decoding will fail, as it expects a 16 bytes-long auth tag. Therefore, I think that raising an error is more appropriate.

Reply:
Ah, so later on we have:

```ruby
          auth_tag = data.slice!(45, 16)
```

And the concern is that there is no framing for length?

Reply:
That's correct.